src/xcode/ENA/ENA/Source/Scenes/Home/HomeBadgeWrapper.swift

```diff
@@ -0,0 +1,53 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+final class HomeBadgeWrapper {
+
+	// MARK: - Init
+
+	init(
+		_ values: [Badges: Int?]
+	) {
+		badgesCount = values
+	}
+
+	convenience init() {
+		self.init([:])
+	}
+
+	// MARK: - Internal
+
+	enum Badges: Int, CaseIterable, Hashable {
+		case unseenTests = 0
+		case riskStateIncreased
+	}
+
+	var updateView: ((String?) -> Void)?
+
+	func update(_ badgeCount: Badges, value: Int?) {
+		badgesCount[badgeCount] = value
+		updateView?(badgeValueString)
+	}
+
+	func reset(_ badgeCount: Badges) {
```

Comment:
I see the issue that you are introducing a second source of truth here for the unseen tests. If I used this method like this `reset(.unseenTests)` it would not reset the `unseenTestsCount` of the corona test service. Maybe it makes sense to go all the way and remove that property and just use the `HomeBadgeWrapper` there as well?